docs/pages/development/build.md

```diff
@@ -34,6 +34,8 @@ This profile has two options set:
 - Setting `developmentClient` to `true` will create a Debug build, which allows the `expo-dev-client` library to allow you to choose the update to load in your app and provide tools to help you develop.
 - Setting `distribution` to "internal" will make a build ready for [internal distribution](/build/internal-distribution).
 
+iOS builds with a developmentClient should always be distributed as "internal". If you set `distribution` to "store" Apple will reject the build with "ITMS-90338: Non-public API usage"
```

Comment:
I think, instead of just targetting iOS builds (and I've read the original issue you posted to get the context), I'm suggesting a change for both TestFlight and Google Play Beta.

 ```suggestion
With `developmentClient` set to `true`, setting the `distribution` to "internal" allows releasing the app for TestFlight and Google Play Beta.
```

@kbrandwijk, can you please take a look if this information is worth adding to docs and is accurate?

Reply:
I think the point here is that you shouldn't be uploading dev-client builds to Apple. I also believe that you need distribution: 'store' to upload to TF, so this rewrite seems incorrect.